jdaviz/app.py

```diff
@@ -1891,6 +1894,94 @@ def _get_viewer_item(self, ref_or_id):
+            item in the data collection that doesn't match ``old_parent`` will be chosen.
+        '''
+        from astropy.wcs.utils import pixel_to_pixel
+
+        if isinstance(old_parent, str):
+            old_parent = self.data_collection(old_parent)
+
+        if isinstance(new_parent, str):
+            new_parent = self.data_collection(new_parent)
+        elif new_parent is None:
+            for data in self.data_collection:
+                if data is not old_parent:
+                    new_parent = data
+                    break
+
+        # Set subset attributes to match a remaining data collection member.
+        for subset_group in self.data_collection.subset_groups:
+            # Flag to skip this
+            if subset_group.subset_state.attributes[0].parent is old_parent:
+                for att in ("att", "xatt", "yatt", "x_att", "y_att"):
+                    if hasattr(subset_group.subset_state, att):
+                        subset_att = getattr(subset_group.subset_state, att)
+                        data_components = new_parent.components
+                        if subset_att not in data_components:
+                            cid = [c for c in data_components if c.label == subset_att.label][0]
+                            setattr(subset_group.subset_state, att, cid)
+
+                # Translate bounds through WCS if needed
+                if (self.config == "imviz" and
+                        self._jdaviz_helper.plugins["Links Control"].link_type == "WCS"):
+                    # Get the correct link to use for translation
+                    roi = subset_group.subset_state.roi
+                    if type(roi) in (CircularROI, CircularAnnulusROI,
+                                     EllipticalROI, TrueCircularROI):
+                        old_xc, old_yc = subset_group.subset_state.center()
+                        # Convert center
+                        x, y = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                              roi.xc, roi.yc)
+                        # Can't use set_center here because CircularAnnulusROI doesn't have it
+                        subset_group.subset_state.move_to(x, y)
+
+                        for att in ("radius", "inner_radius", "outer_radius",
+                                    "radius_x", "radius_y"):
+                            # Hacky way to get new radii with point on edge of circle
+                            # Do we need to worry about using x for the radius conversion for
+                            # radius_y if there is distortion?
+                            r = getattr(roi, att, None)
+                            if r is not None:
+                                dummy_x = old_xc + r
+                                x2, y2 = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                                        dummy_x, old_yc)
+                                new_radius = np.abs(x2 - x)
```

Comment:
Would it be better to be more consistent with how "strict circle" is approximating a radius from something that might no longer be a circle?

https://github.com/glue-viz/glue-jupyter/blob/0b1a3fb0399dd445061101100a166309fe2d9f0b/glue_jupyter/bqplot/common/tools.py#L404

Reply:
in theory this won't be an issue once the rotation PR gets in (since reference will be enforced to be on-sky when WCS linked)